### DISCUSSION TITLE
Information about necessary/optional Kernel Modules and their purpose

### DISCUSSION THREAD

**Discussion by mluhmann** [containers/podman#24293] [Q&A]
Hi everyone,
I am in a project building our own Linux including the kernel, based on the Yocto project. Unfortunately the podman (and related software) recipes do not come with any kernel module suggestions. We figured out a lot of kernel modules by trial and error or based of on of the shelf Linux OS. But now we are facing some issues and I'm asking for help to maybe optimize and remove unnecessary modules. 
I wasn't able to find any information about this, just some check in the podman tests.
I am currently running podman, rootless with cgroups v2 on a kernel 6.1. I need especially information about modules for cgroups controller, because this is probably the root cause of our issues. 
Here are the currently active cgroup modules:

- CONFIG_CGROUPS
- CONFIG_MEMCG
- CONFIG_BLK_CGROUP
- CONFIG_CGROUP_SCHED
- CONFIG_CGROUP_PIDS
- CONFIG_CGROUP_RDMA
- CONFIG_CGROUP_FREEZER
- CONFIG_CGROUP_HUGETLB
- CONFIG_CPUSETS
- CONFIG_CGROUP_DEVICE
- CONFIG_CGROUP_CPUACCT
- CONFIG_CGROUP_PERF
- CONFIG_CGROUP_BPF
- CONFIG_CGROUP_MISC
- CONFIG_CGROUP_DEBUG

Further more, I try to get nftables and pasta running with podman, but missing out on some modules as well.
Any help or hint where to find or how to get this information would be great.

Thanks!

**Comment by rhatdan** [comment]
@giuseppe PTAL

**Comment by giuseppe** [comment]
what error are you hitting?
